### PR TITLE
Venom: wired-mac-auth scenario

### PR DESCRIPTION
**Pull request by nqb**
# Description
Test a complete wired-mac-auth scenario using Venom

# Impacts
Integration tests

# Delete branch after merge
YES

### CODE REVIEW

File: t/venom/pfservers/wired_mac_auth/00_enable_node_cleanup_task.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      {
+        "status": "enabled"
+      }
+    headers:
+      "Authorization": "{{.pfserver_token}}"
+      "Content-Type": "application/json"
+    assertions:
+      - result.statuscode ShouldEqual 200
+    
+- name: change_delete_window_to_1M
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/00_enable_node_cleanup_task.yml]: no uppercase:

```suggestion
- name: change_delete_window_to_1m
```

File: t/venom/pfservers/wired_mac_auth/05_create_node01.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+testcases:
+- name: create_node01
+  steps:
+  - type: http
+    method: POST
+    url: '{{.pfserver_webadmin_url}}/api/v1/nodes'
+    ignore_verify_ssl: true
+    body: >-
+      {
+        "category_id": "30",
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/05_create_node01.yml]: This need to be a variable, we will need @Ieno77 commits.

File: t/venom/vars/all.yml [hunk 3]
```diff
@@ -72,3 +74,6 @@ dot1x_eap_peap.profiles.wired.id: catch_dot1x_wired_eap_peap_mschapv2
 dot1x_eap_peap.profiles.wired.description: 802.1X wired EAP-PEAP-MSCHAPv2
 dot1x_eap_peap.profiles.wired.filters.connection_type: Ethernet-EAP
 dot1x_eap_peap.profiles.wired.filters.connection_sub_type: MS-EAP-Authentication
+mac_authentication.profiles.wired.id: catch_wired_mac_authentication
+mac_authentication.profiles.wired.description: Wired Mac authentication with unreg_on_accounting_stop
+mac_authentication.profiles.wired.filters.connection_type: Ethernet-NoEAP
```
Comment (nqb) [inline t/venom/vars/all.yml]: All variables related to this scenario should be prefixed by `wired_mac_auth` in place of `mac_authentication`

File: t/venom/vars/all.yml [hunk 4]
```diff
@@ -29,6 +29,8 @@ switch01_mgmt_ip: 172.17.17.201
 
 # nodes variables
 node01_mgmt_ip: 172.17.17.251
+node01_unreg_date: '2020-04-20 13:37:00'
```
Comment (nqb) [inline t/venom/vars/all.yml]: We should use a dynamic variable.

File: t/venom/pfservers/wired_mac_auth/10_create_connection_profile.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        "redirecturl": null,
+        "reuse_dot1x_credentials": null,
+        "root_module": "default_policy",
+        "scans": null,
+        "self_service": null,
+        "sms_pin_retry_limit": 0,
+        "sms_request_limit": 0,
+        "sources": null,
+        "status": "enabled",
+        "unreg_on_acct_stop": "enabled",
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/10_create_connection_profile.yml]: Create a variable for this setting like `wired_mac_auth.profiles.unreg_on_acct_stop: enabled`

File: t/venom/pfservers/wired_mac_auth/teardown/10_verify_locationlog_closed.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+name: Check if node01 latest locationlog is closed
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/teardown/10_verify_locationlog_closed.yml]: `check_locationlog_closed` seems more consistent thant `verify`

File: t/venom/pfservers/wired_mac_auth/teardown/15_delete_nod01_with_pfmon_node_cleanup.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+name: Delete node01 by running pfmon node_cleanup task
+version: 2
+testcases:
+- name: delete_nod01_with_pfmon_node_cleanup
+  steps:
+  - script: /usr/local/pf/bin/pfcmd pfmon node_cleanup
+    assertions:
+    - result.code ShouldEqual 0
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/teardown/15_delete_nod01_with_pfmon_node_cleanup.yml]: This assertion is not necessary (default behavior)

File: t/venom/pfservers/wired_mac_auth/teardown/25_enable_node_cleanup_task.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      {
+        "status": "disabled"
+      }
+    headers:
+      "Authorization": "{{.pfserver_token}}"
+      "Content-Type": "application/json"
+    assertions:
+      - result.statuscode ShouldEqual 200
+    
+- name: change_delete_window_to_1M
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/teardown/25_enable_node_cleanup_task.yml]: no uppercase:
```suggestion
- name: change_delete_window_to_1m
```

File: t/venom/vars/all.yml (hunk 4 above)
Comment (nqb) [reply t/venom/vars/all.yml]:
```
- name: get_unreg_date
  steps:
  - type: exec
    script: "date '+%Y-%m-%d %H:%M:%S' --date='next minute'"
    vars:
      node01_unreg_date:
        from: result.systemout   
```

File: t/venom/pfservers/wired_mac_auth/teardown/20_check_node01_deleted.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                  "value": "{{.node01_eth1_mac_address}}"
+                }
+              ]
+            }
+          ]
+        },
+        "sort": [
+          "mac"
+        ]
+      }
```
Comment (nqb) [inline t/venom/pfservers/wired_mac_auth/teardown/20_check_node01_deleted.yml:47]: add `retry` and `delay` statement to be sure node is deleted.

File: t/venom/vars/all.yml [hunk 10]
```diff
@@ -77,3 +77,9 @@ dot1x_eap_peap.profiles.wired.filters.connection_sub_type: MS-EAP-Authentication
 mac_authentication.profiles.wired.id: catch_wired_mac_authentication
 mac_authentication.profiles.wired.description: Wired Mac authentication with unreg_on_accounting_stop
 mac_authentication.profiles.wired.filters.connection_type: Ethernet-NoEAP
+
+################################################################################
+## wired_mac_auth variables
+#################################################################################
+role_headless_device: headless_device
+vlan_headless_device: 20
```
Comment (lzammit) [inline t/venom/vars/all.yml]: @nqb don't we need to choose an ID as well or we do fetch it automatically after creating it ?

Comment (nqb) [reply t/venom/vars/all.yml]: `vlan_headless_device` should be: 100 because that's the production VLAN we used in CI environment.

**Comment by nqb** [comment]
Scenario is working. I'm running a last pipeline on Debian and CentOS and I will merge if all is fine.

@lzammit, if you want to make a latest review, you're welcome.

**Pull request merged**